psi4/driver/driver_cbs.py

```diff
@@ -917,65 +1110,272 @@ def return_energy_components():
+    """ A helper function to determine default extrapolation alpha. Currently only
+    used by :py:func:`_interpret_cbs_inputs` to get default alphas for DFA, as both
+    "scf" and "corl" types have their default alpha set.
+
+
+    Parameters
+    ----------
+    xtpl_type
+        {'scf', 'corl', 'fctl', 'dh', None}
+        Extrapolation type: 'scf' and 'fctl' for the total energy, 'corl' and 'dh'
+        for just the correlation component, None if no extrapolation requested.
+    bdata
+        Names and zetas of the basis sets to extrapolate from.
+    funcname
+        Functional name for DFT extrapolation.
+
+    Returns
+    -------
+    alpha : Union[float, None]
+        Extrapolation alpha to be used.
+    """
+    nbasis = len(bdata[0])
+    if nbasis == 1:
+        return None
+    elif nbasis == 2:
+        if xtpl_type == "scf":
+            return 1.63  # this is the scf_xtpl_helgaker_2 default
+        elif xtpl_type == "corl":
+            return 3.00  # this is the corl_xtpl_helgaker_2 default
+        elif xtpl_type in ["fctl", "dh", "dft"]:
+            return _get_dfa_alpha(xtpl_type, bdata, funcname)
+        else:
+            raise ValidationError(
+                f"Stage treatment must be one of ['scf','corl','fctl','dh',"
+                f"'dft'], not '{xtpl_type}'"
+            )
+    elif nbasis == 3:
+        return None  # we don't need alpha for _3 methods
+
+
+def _interpret_cbs_inputs(cbs_metadata: List, molecule: Union[qcdb.molecule.Molecule, core.Molecule]) -> list:
     """ A helper function which validates the ``cbs_metadata`` format,
     expands basis sets, and provides sensible defaults for optional arguments.
 
+    There are currently three ways to run a CBS calculation:
+
+    1) by specifying arguments, such as ``scf_wfn``, ``corl_basis``, or ``delta_scheme``
+    2) by using the "method/basis+delta" syntax
+    3) by explicitly passing ``cbs_metadata`` as one of the arguments
+
+    These three are interpreted into a single, internal specification here. For the
+    description of the user-facing interfaces see :py:func:`~psi4.cbs`.
+
+    The main role of this function is to interpret the user-provided arguments,
+    infer any missing stages (if possible) such as the SCF stage if the user
+    provides "MP2/cc-pv[dt]z" or the split stages if user requests an extrapolation
+    of a DFA.
+
+    The resulting ``metadata`` :class:`(list[dict])` contains the list of all stages,
+    with possible entries in each of the dicts being:
+
+     - ``"wfn"`` :class:`(str)`: method name for upper level of theory (l.o.t.)
+     - ``"basis"`` :class:`(tuple)`: definition of the basis sets for the upper
+       l.o.t., provided as an "expanded" tuple of names and zeta levels
+     - ``"options"`` :class:`(dict)`: specification of additional Psi4 options to
+       switch on in the upper l.o.t.
+     - ``"component"`` :class:`(str)`: component of the total E/G/H to use for
```

Comment:
How can I possibly select a _component_ of the energy as an upper level of theory? Treating the dispersion as an upper level of theory and non-local as the lower level of theory is nonsense.

Reply:
I don't think it's such a stupid idea.

Reply:
You'll have to explain why. My example would give + DISP - NL. Why would you want to do that?

Reply:
There's a use case if you want to do `"dft"` - `"fctl"`. The component of the lower wfn needs to be specified, even though 99% of the time it will be the same as upper, but it makes sense to expose it.